repo/packages/A/arangodb/0/config.json

```diff
@@ -0,0 +1,86 @@
+{
+  "type": "object",
+  "properties": {
+
+    "mesos/master": {
+      "description": "The URL of the Mesos master. The format is a comma-delimited list of hosts like zk://host1:port,host2:port/mesos. If using ZooKeeper, pay particular attention to the leading zk:// and trailing /mesos! If not using ZooKeeper, standard URLs like http://localhost are also acceptable.",
+      "type": "string",
+      "default": "zk://master.mesos:2181/mesos"
+    },
+
+    "arangodb/id": {
+      "description": "Identifier for the cluster.",
+      "type": "string",
+      "default": "arangodb"
+    },
+
+    "arangodb/zk": {
+      "description": "ZooKeeper URL for storing state. Format: zk://host1:port1,host2:port2,.../path. Please note that the id is appended to the path.",
+      "type": "string",
+      "default": "zk://master.mesos:2181/arangodb"
+    },
+
+    "arangodb/webui-host": {
+      "description": "For debugging: use this host for WEBUI request.",
+      "type": "string",
+      "default": ""
+    },
+
+    "arangodb/framework-port": {
+      "description": "Framework port, use '0' to let Mesos select a port for you.",
+      "type": "integer",
+      "default": 0
+    },
+
+    "arangodb/volume-path": {
+      "description": "Path for persistent data (until persistent volumes are available).",
+      "type": "string",
+      "default": "/tmp"
+    },
+
+    "arangodb/mode": {
+      "description": "Mode for framework, possible values: \"standalone\", \"cluster\"",
+      "type": "string",
+      "default": "cluster"
+    },
+
+    "arangodb/minimal-resources-agent": {
+      "description": "Minimal mesos resources for an agent",
+      "type": "string",
+      "default": "mem(*):512;cpus(*):0.2;disk(*):512"
+    },
+
+    "arangodb/minimal-resources-coordinator": {
+      "description": "Minimal mesos resources for a coordinator",
+      "type": "string",
+      "default": "mem(*):1024;cpus(*):1;disk(*):1024"
+    },
+
+    "arangodb/minimal-resources-dbserver": {
+      "description": "Minimal mesos resources for a DBserver",
+      "type": "string",
+      "default": "mem(*):1024;cpus(*):1;disk(*):1024"
+    },
+
+    "arangodb/nr-agents": {
+      "description": "Initial number of agents",
+      "type": "integer",
+      "default": 1
+    },
+
+    "arangodb/nr-dbservers": {
+      "description": "Initial number of DBservers",
+      "type": "integer",
+      "default": 2
+    },
+
+    "arangodb/nr-coordinators": {
+      "description": "Initial number of coordinators",
+      "type": "integer",
+      "default": 1
+    }
+  },
+
+  "required": [
```

Comment:
I am not sure I completely understand this. What is the exact meaning of the "required" attribute in this file? I thought it is what the user has to specify, and since we provide defaults for everything, nothing is required. Is this thinking flawed?
I think I do understand the marathon business: we have to make cpu, mem etc. for the framework configurable here and then set them accordingly in marathon.json?